src/Agent.Worker/Release/Artifacts/BuildArtifact.cs

```diff
@@ -236,8 +236,20 @@ private bool Match(ServerBuildArtifact buildArtifact, ArtifactDefinition artifac
                     DownloadBufferSize = executionContext.Variables.Release_Download_BufferSize ?? ContainerFetchEngineDefaultOptions.DownloadBufferSize
                 };
 
+                int fetchEngineTimeoutSeconds;
+                if (!int.TryParse(Environment.GetEnvironmentVariable("VSTS_FETCHENGINE_TIMEOUT") ?? string.Empty, out fetchEngineTimeoutSeconds))
```

Comment:
@stephenmichaelf my fault, i forgot v2 agent use `VSTS_HTTP_TIMEOUT` even for file container upload. I agree with @GitHubSriramB, let's just use `VSTS_HTTP_TIMEOUT`.  
check this: https://github.com/Microsoft/vsts-agent/blob/master/src/Agent.Worker/Build/FileContainerServer.cs#L43